### PR TITLE
feat: ask to save wishlist changes when user leave

### PR DESCRIPTION
**Pull request by asvishnyakov**
Description:
https://virtocommerce.atlassian.net/browse/ST-3339

--

QA-test:

Demo-test:

Download artifact URL: https://vc3prerelease.blob.core.windows.net/packages/vc-theme-b2b-vue-1.37.0-pr-685-5360-53603d24.zip

### CODE REVIEW

File: client-app/pages/account/list-details.vue [hunk 1]
```diff
@@ -223,31 +225,37 @@ async function addAllListItemsToCart(): Promise<void> {
...
-    props: {
-      variant: "info",
-      noIcon: true,
-      title: t("common.labels.save_changes"),
-      text: t("common.messages.save_new_product_quantity"),
-      onConfirm: async () => {
-        closeDialog();
-
-        await updateWishlistItemsQuantities(payload);
+async function openSaveChangesModal(): Promise<boolean> {
```
Comment (NaMax66) [inline client-app/pages/account/list-details.vue:239]: no need "async" if no "await"

File: client-app/pages/account/list-details.vue [hunk 2]
```diff
@@ -223,31 +225,37 @@ async function addAllListItemsToCart(): Promise<void> {
...
+  return new Promise<boolean>((resolve) => {
+    const closeDialog = openPopup({
+      component: VcConfirmationDialog,
+      props: {
+        variant: "info",
+        noIcon: true,
+        title: t("pages.account.list_details.save_changes"),
+        text: t("pages.account.list_details.save_changes_message"),
+        onConfirm: async () => {
```
Comment (NaMax66) [inline client-app/pages/account/list-details.vue:248]: same

File: client-app/pages/account/list-details.vue (hunk 1 above)
Comment (asvishnyakov) [reply client-app/pages/account/list-details.vue]: done

File: client-app/pages/account/list-details.vue (hunk 2 above)
Comment (asvishnyakov) [reply client-app/pages/account/list-details.vue]: done

**Review by NaMax66 — APPROVED** [review]

**Pull request merged**